videostore/utils.c

```diff
@@ -1,24 +1,50 @@
 #include "utils.h"
 
-int get_video_duration(int64_t *duration,   // OUT
-                       const char *filename // IN
-) {
+int get_video_info(VideoInfo *info, const char *filename)
+{
+    info->duration = 0;
+    info->width = 0;
+    info->height = 0;
+    info->codec[0] = '\0';
+
     AVFormatContext *fmt_ctx = NULL;
     int ret;
+
     if ((ret = avformat_open_input(&fmt_ctx, filename, NULL, NULL)) < 0) {
         return ret;
     }
     if ((ret = avformat_find_stream_info(fmt_ctx, NULL)) < 0) {
         avformat_close_input(&fmt_ctx);
-    return ret;
+        return ret;
+    }
+
+    info->duration = fmt_ctx->duration;
+    if (info->duration == AV_NOPTS_VALUE) {
+        avformat_close_input(&fmt_ctx);
+        return VIDEO_STORE_VIDEO_INFO_RESP_ERROR;
+    }
+
+    for (unsigned i = 0; i < fmt_ctx->nb_streams; i++) {
+        AVStream *st = fmt_ctx->streams[i];
+        if (st->codecpar->codec_type == AVMEDIA_TYPE_VIDEO) {
```

Comment:
Assuming we want to fetch info for the first video stream found in the video file.

Reply:
Could there be multiple? I dont' quite understand this assumption

Reply:
Technically, an mp4 file can have multiple video streams inside of it. But for our use case, it is safe to assume that segment files will only have one video track.